audio/src/passthrough_decoder.rs

```diff
@@ -131,29 +133,28 @@ impl<R: Read + Seek> AudioDecoder for PassthroughDecoder<R> {
 
             // NB: we don't handle multiple streams (change of serial)
             let stream_serial = pck.stream_serial();
-            let absgp_page = pck.absgp_page();
+            let pckgp_page = pck.absgp_page();
 
             // set granule if unknown
-            if self.absgp_page == u64::MAX {
-                self.absgp_page = absgp_page;
+            if self.lastgp_page.is_none() {
+                self.lastgp_page = Some(pckgp_page);
```

Comment:
You could use something like 
```rust
let lastgp_page = *self.lastgp_page.get_or_insert(pckgp_page);
```

to avoid the `unwrap`s below.

Reply:
done